src/braket/circuits/synthesis/two_qubit_decomposition.py

```diff
@@ -0,0 +1,580 @@
+        )
+
+        self.phase = g1 * g2 * np.exp(1j * kak_4vector[0])
+
+        self.su2 = [u1, u2, u3, u4]
+        self.canonical_vector = kak_4vector[1:]
+
+    def __repr__(self):
+
+        str_u = []
+        for i in range(4):
+            str_u.append(np.array2string(self.su2[i]).replace("\n", "\n     "))
+
+        kak_str = (
+            "TwoQubitDecomposition(\n"
+            + "  U = (u1 ⊗ u2) · exp(i(v0·XX + v1·YY+v2·ZZ))·(u3 ⊗ u4)\n"
+            + f"  global phase: {self.phase},\n"
+            + f"  canonical vector v: {self.canonical_vector},\n"
+            + f"  u1: {str_u[0]},\n"
+            + f"  u2: {str_u[1]},\n"
+            + f"  u3: {str_u[2]},\n"
+            + f"  u4: {str_u[3]}\n)"
+        )
+        return kak_str
+
+    @property
+    def unitary(self):
+        """
+        The unitary matrix of this decomposition.
+        """
+        return self.U
+
+    def pretty_print(self) -> str:
+        """
+        Helper function for printing detailed information of the
+        decomposition.
+        """
+
+        str_u = []
+        for i in range(4):
+            str_u.append(np.array2string(self.su2[i], precision=3).replace("\n", "\n     "))
+        kak_str = (
+            "TwoQubitDecomposition(\n"
+            + "  U = (u1 ⊗ u2) · exp(i(v0·XX + v1·YY+v2·ZZ))·(u3 ⊗ u4)\n"
+            + f"  global phase: {np.array2string(self.phase, precision=3)},\n"
+            + f"  canonical vector: {np.array2string(self.canonical_vector, precision=3)},\n"
+            + f"  u1: {str_u[0]},\n"
+            + f"  u2: {str_u[1]},\n"
+            + f"  u3: {str_u[2]},\n"
+            + f"  u4: {str_u[3]}\n)"
+        )
+        print(kak_str)
+        return kak_str
+
+    def count_num_cnots(self):
+        """
+        Find the number of CNOTs required to
+        synthesize self.U.
+
+        Returns:
+            num_cnots (int): The number of CNOTs required.
+        """
+
+        gamma_inv = gamma_invariants(self.U, atol=self.atol, rtol=self.rtol, validate_input=False)
+
+        if np.allclose(
+            gamma_inv, _ZERO_CNOT_GAMMA_INV1, atol=self.atol, rtol=self.rtol
+        ) or np.allclose(gamma_inv, _ZERO_CNOT_GAMMA_INV2, atol=self.atol, rtol=self.rtol):
+            return 0
+
+        if np.allclose(gamma_inv, _ONE_CNOT_GAMMA_INV, atol=self.atol, rtol=self.rtol):
+            return 1
+
+        if np.allclose(np.imag(gamma_inv), _TWO_CNOT_GAMMA_INV, atol=self.atol, rtol=self.rtol):
+            return 2
+
+        else:
+            return 3
+
+    def to_circuit(self, qubits: Union[QubitSet, Sequence[int]] = [0, 1]):
+        """
+        Build the synthesized circuit.
+
+        Args:
+            qubits (Union[QubitSet, Sequence[int]]): The qubit set to build the circuit on.
+        """
+
+        return self.to_cnot_circuit(qubits=qubits)
+
+    def to_cnot_circuit(self, qubits: Union[QubitSet, Sequence[int]] = [0, 1]) -> Circuit:
```

Comment:
Can to_cnot_circuit, _to_zero_cnot_circuit, _to_one_cnot_circuit, _to_two_cnot_circuit, _to_three_cnot_circuit all be marked private? It seems simpler to just call to_circuit, and handle the decision of what type of circuit is generated for the customer.